### ISSUE TITLE
[iOS] Cannot Compile Due to Problem OpenSSL

### ISSUE THREAD

**Issue by computerion** [FreeRDP/FreeRDP#3456]
I'm trying to follow this README on my El Capitan MBP for building the `.xcodeproj`:

https://github.com/FreeRDP/FreeRDP/blob/master/docs/README.ios

Running this gives me the following error:
```
-- Finding required feature OpenSSL for cryptography (encryption, certificate validation, hashing functions)
CMake Error at /usr/local/Cellar/cmake/3.4.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_LIBRARIES
  OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.4.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindOpenSSL.cmake:317 (find_package_handle_standard_args)
  cmake/FindFeature.cmake:16 (find_package)
  CMakeLists.txt:705 (find_feature)
```

I definitely have `openssl` installed:
```
$ which openssl
/usr/local/bin/openssl
```

So I tried to manually set the environment variables with:
```
cmake -DCMAKE_TOOLCHAIN_FILE=cmake/iOSToolchain.cmake -DPKG_CONFIG_EXECUTABLE=/usr/local/bin/pkg-config -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DOPENSSL_LIBRARIES=/usr/local/opt/openssl/lib -DOPENSSL_LIBRARIES=/usr/local/opt/openssl/include -DPKG_CONFIG_PATH=/usr/local/opt/pkg-config -GXcode
```

Still running into the same issue however. Any tips on how to get around this? Thanks!

**Comment by akallabeth** [comment]
@computerion You need OpenSSL for iOS, not your host OS ;) Use the script referenced in the README to compile it for all necessary architectures. (note: the script on master may be out of date, an updated version can be found here: https://github.com/akallabeth/FreeRDP/blob/6f748a9b774b0b09c7f3059213610e0a228827bc/scripts/OpenSSL-DownloadAndBuild.command

**Comment by computerion** [comment]
Thanks @akallabeth
Using your new script solved the problem. However I should note that I was still forced to set the
`OPENSSL_ROOT_DIR` variables in order to get it work properly. This should probably be updated on the doc perhaps?

**Comment by akallabeth** [comment]
@computerion Thx, found the bug. the script does not properly copy the openssl headers to ```external/openssl/include```

**Comment by computerion** [comment]
Awesome thanks a lot for the help!

**Comment by akallabeth** [comment]
Fixed with pull #3450
